### PR TITLE
bcm283x: Enable DMA functions.

### PR DESCRIPTION
**Pull request by simokawa**
bcm283x DMA improvements #172
- Enable DMA driven bitstream.
- Start DMA writing PWM FIFO.
- Round off PWM duty.

### CODE REVIEW

File: host/bcm283x/gpio.go [hunk 1]
```diff
@@ -479,11 +480,57 @@ func (p *Pin) PWM(duty gpio.Duty, period time.Duration) error {
...
 	}
-
+	p.usingClock = true
 	p.setFunction(f)
 	return nil
 }
 
+// StreamIn implements gpiostream.PinIn.
+//
+// DMA driven StreamOut is avaiable for GPIO0 to GPIO31 pin and the resolution is
```
Comment (maruel) [inline host/bcm283x/gpio.go]: available*
"the resolution is 200kHz" maybe "maximum" ?
"kHz" is the right way to write the unit. I'll do a separate commit to fix the inconsistencies in the code base. :)

File: host/bcm283x/gpio.go [hunk 2]
```diff
@@ -479,11 +480,57 @@ func (p *Pin) PWM(duty gpio.Duty, period time.Duration) error {
...
+	}
+	return nil
+}
+
+// StreamOut implements gpiostream.PinOut.
+//
+// PCM/I2S driven StreamOut is available for GPIO21 pin. The resolution is up to
+// 250MHz.
+//
+// For GPIO0 to GPIO31 except GPIO21 pin, DMA driven StreamOut is avaiable and
```
Comment (maruel) [inline host/bcm283x/gpio.go]: available*
same comment about maximum and kHz.

**Comment by maruel** [comment]
Sorry you'll have to rebase and I was away for the last days.

File: host/bcm283x/gpio.go (hunk 1 above)
Comment (simokawa) [reply host/bcm283x/gpio.go]: Done.

File: host/bcm283x/gpio.go (hunk 2 above)
Comment (simokawa) [reply host/bcm283x/gpio.go]: Done.

**Comment by simokawa** [comment]
I rebased on the master.

**Comment by simokawa** [comment]
I cleaned up my forked branch and rebased again.

**Comment by maruel** [comment]
gohci

**Pull request merged**